### PR TITLE
Adding threat modeling example to security documentation

### PR DESCRIPTION
**Pull request by anandchugh**
# Pull Request Template

## What are you trying to address
Adding example of threat model which can act as reference.

- Describe the current behavior that you are modifying, and link to issue number
-Here is [link](https://github.com/microsoft/code-with-engineering-playbook/issues/890) to the issue.
- If you don't have an issue, browse through existing issues to see if this is already tracked as an issue, to assign yourself to the issue and also verify that no one else is already working on the issue.

## Description of new changes
Adding sample threat model and added reference to this document in main README.md file.
- Write a detailed description of all changes - and if appropriate, why they are needed

## For all pull requests

- [X] Link to the issue you are solving (so it gets closed)
- [X] Label the pull request with the appropriate area(s)
- [X] Assign potential reviewers (you may also want to contact them on Teams to ensure timely reviews)
- [X] Assign the project - Engineering Playbook Backlog
- [X] Assign the pull request to yourself

### CODE REVIEW

File: docs/security/threat-modelling.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Threat Modeling
+
+Threat modeling is an effective way to help secure your systems, applications, networks, and services. It's an systematic approach that identifies potential threats and recommendations to help reduce risk and meet security objectives earlier in the development lifecycle.
```
Comment (shiranr) [inline docs/security/threat-modelling.md:3]:
```suggestion
Threat modeling is an effective way to help secure your systems, applications, networks, and services. It's a systematic approach that identifies potential threats and recommendations to help reduce risk and meet security objectives earlier in the development lifecycle.
```

File: docs/security/threat-modelling.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+## Threat Modeling Phases
+
+1. *Diagram*
```
Comment (shiranr) [inline docs/security/threat-modelling.md:7]: Just a suggestion: Maybe we can add some project example of this flow?

File: docs/security/threat-modelling.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# Threat Modeling
+
+Threat modeling is an effective way to help secure your systems, applications, networks, and services. It's an systematic approach that identifies potential threats and recommendations to help reduce risk and meet security objectives earlier in the development lifecycle.
+
+## Threat Modeling Phases
+
+1. *Diagram*  
+    Capture all requirements for your system and create a data-flow diagram
+2. *Identify*  
+    Apply a threat-modeling framework to the data-flow diagram and find potential security issues. Here we can use STRIDE framework to identify the threats.
```
Comment (shiranr) [inline docs/security/threat-modelling.md:10]: Perhaps add a link?

File: docs/security/threat-modelling.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+3. *Mitigate*  
+    Decide how to approach each issue with the right combination of security controls.  
+4. *Validate*  
+    Verify requirements are met, issues are found, and security controls are implemented.
+
+More details about these phases can be found at [Threat Modeling Security Fundamentals.](https://learn.microsoft.com/en-us/training/paths/tm-threat-modeling-fundamentals/)
+
+## Threat Modeling Example
+
+   [Here is an example](./threat-modelling-example.md) of threat modeling document which is talks about the architecture and different phases involved in the threat modeling. This document can be used as reference template for creating threat modeling documents.
```
Comment (shiranr) [inline docs/security/threat-modelling.md:20]:
```suggestion
   [Here is an example](./threat-modelling-example.md) of threat modeling document which talks about the architecture and different phases involved in the threat modeling. This document can be used as reference template for creating threat modeling documents.
```

**Review by shiranr — COMMENTED** [review]
Thanks for contributing @anandchugh, I left some small comments and suggestions.

File: docs/security/threat-modelling.md (hunk 2 above)
Comment (anandchugh) [reply docs/security/threat-modelling.md:7]: Add the link to example document.

> Just a suggestion: Maybe we can add some project example of this flow?

Adding link to example document

File: docs/security/threat-modelling.md (hunk 1 above)
Comment (anandchugh) [reply docs/security/threat-modelling.md]: Fixed

File: docs/security/threat-modelling.md (hunk 3 above)
Comment (anandchugh) [reply docs/security/threat-modelling.md]: Added link to STRIDE framework

**Comment by anandchugh** [comment]
> Thanks for contributing @anandchugh, I left some small comments and suggestions.

Thanks for review, updated with fixes.

**Review by meulta — APPROVED** [review]
Thanks for the contribution!

File: docs/automated-testing/e2e-testing/README.md [hunk 5]
```diff
@@ -137,7 +137,6 @@ Run tests in parallel or on a Selenium Grid with built-in Selenium WebDriver. Ra
 
 **Ranorex Studio** tests also integrate with Azure DevOps (AzDO), which can be run as part of a build pipeline in AzDO.
 
-[Ranorex Studio Website](https://www.ranorex.com/free-trial/?utm_source=Google&utm_medium=cpc&utm_id=7015a000001ohs3AAA&utm_campaign=14484569941&utm_source=g_c&utm_medium=cpc&utm_content=588982669352&utm_term=p_ranorex%20studio&adgroupid=132316228251&gclid=EAIaIQobChMIsoqzgdzD-AIVygMGAB1pHQrpEAAYASAAEgKed_D_BwE)
```
Comment (sbohlen) [inline docs/automated-testing/e2e-testing/README.md:140]: GENERAL RECOMMENDATION: remove files having changes due exclusively to auto-formatting/whitespace edits from this PR and submit as part of a separate PR (focused on misc. whitespace/formatting updates) in order to reduce the 'noise' in this PR.

File: docs/machine-learning/ml-feasibility-study.md [hunk 6]
```diff
@@ -1,10 +1,10 @@
 # Feasibility Studies
 
 The main goal of feasibility studies is to assess whether it is feasible to solve the problem satisfactorily using ML with the available data. We want to avoid investing too much in the solution before we have:
-*	Sufficient evidence that a solution would be the best technical solution given the business case
```
Comment (sbohlen) [inline docs/machine-learning/ml-feasibility-study.md:4]: GENERAL RECOMMENDATION: remove files having changes due exclusively to auto-formatting/whitespace edits from this PR and submit as part of a separate PR (focused on misc. whitespace/formatting updates) in order to reduce the 'noise' in this PR.

File: docs/observability/tools/OpenTelemetry.md [hunk 7]
```diff
@@ -92,7 +92,7 @@ Java OpenTelemetry instrumentation provides another way to integrate with Azure
 
 When configuring this option, the Applications Insights Agent file is added when executing the application. The `applicationinsights.json` configuration file must be also be added as part of the applications artifacts. Pay close attention to the preview section, where the `"openTelemetryApiSupport": true,` property is set to true, enabling the agent to intercept OpenTelemetry telemetry created in the application code pushing it to Azure Monitor.
 
-OpenTelemetry Java Agent instrumentation supports many [libraries and frameworks and application servers](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/supported-libraries.md#supported-libraries-frameworks-application-servers-and-jvms). Application Insights Java Agent [enhances](https://docs.microsoft.com/en-us/azure/azure-monitor/app/java-in-process-agent#auto-instrumentation) this list.
```
Comment (sbohlen) [inline docs/observability/tools/OpenTelemetry.md:95]: GENERAL RECOMMENDATION: remove files having changes due exclusively to auto-formatting/whitespace edits from this PR and submit as part of a separate PR (focused on misc. whitespace/formatting updates) in order to reduce the 'noise' in this PR.

File: docs/security/threat-modelling.md [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+Threat modeling is an effective way to help secure your systems, applications, networks, and services. It's a systematic approach that identifies potential threats and recommendations to help reduce risk and meet security objectives earlier in the development lifecycle.
+
+## Threat Modeling Phases
+
+1. *Diagram*  
+    Capture all requirements for your system and create a data-flow diagram
+2. *Identify*  
+    Apply a threat-modeling framework to the data-flow diagram and find potential security issues. Here we can use [STRIDE framework](https://learn.microsoft.com/en-us/training/modules/tm-use-a-framework-to-identify-threats-and-find-ways-to-reduce-or-eliminate-risk/1b-threat-modeling-framework) to identify the threats.  
+3. *Mitigate*  
+    Decide how to approach each issue with the right combination of security controls.
```
Comment (sbohlen) [inline docs/security/threat-modelling.md:12]: nit: consider changing this to (emphasis mine):

> Decide how to approach each issue with the ~right~ **appropriate** combination of security controls.

File: docs/security/threat-modelling.md [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+    Decide how to approach each issue with the right combination of security controls.  
+4. *Validate*  
+    Verify requirements are met, issues are found, and security controls are implemented.
+
+Example of these phases is covered in the [threat modelling example.](./threat-modelling-example.md)  
+More details about these phases can be found at [Threat Modeling Security Fundamentals.](https://learn.microsoft.com/en-us/training/paths/tm-threat-modeling-fundamentals/)
+
+## Threat Modeling Example
+
+   [Here is an example](./threat-modelling-example.md) of threat modeling document which talks about the architecture and different phases involved in the threat modeling. This document can be used as reference template for creating threat modeling documents.
```
Comment (sbohlen) [inline docs/security/threat-modelling.md:21]: nit: change to either (emphasis mine)...

> [Here is an example](./threat-modelling-example.md) of **a** threat modeling document which talks about

or 

> [Here is an example](./threat-modelling-example.md) ~of~ threat modeling document which talks about

**Review by sbohlen — CHANGES_REQUESTED** [review]
This looks really great -- thanks so much for the contribution! One you separate the several whitespace-only-modified files into a separate PR, you should be good to go with this one!

File: docs/automated-testing/e2e-testing/README.md (hunk 5 above)
Comment (anandchugh) [reply docs/automated-testing/e2e-testing/README.md]: This is fixed by latest merge from main branch.

File: docs/observability/tools/OpenTelemetry.md (hunk 7 above)
Comment (anandchugh) [reply docs/observability/tools/OpenTelemetry.md]: This is fixed by latest merge from main branch.

**Review by sbohlen — APPROVED** [review]
Great work; really helpful addition to the content!

**Comment by sbohlen** [comment]
merging accordingly
